CMakeLists.txt

```diff
@@ -58,8 +61,11 @@ include(${CMAKE_CURRENT_SOURCE_DIR}/cmake-modules/az_version.cmake)
 # sub-projects
 # add_subdirectory(sdk/core/performance-stress)
 add_subdirectory(sdk/core/azure-core)
-if(BUILD_CURL_TRANSPORT)
-    add_subdirectory(sdk/core/azure-core/test/e2e) # will work only if BUILD_CURL_TRANSPORT=ON
+
+# There is always a transport adapter added by default.
+if(BUILD_TESTING)
+    add_subdirectory(sdk/core/azure-core/test/e2e)
```

Comment:
What  does the transport adapter comment have to do with this code: `if (BUILD_TESTING)`

Reply:
the comment justify removing the previous if() condition which would've been added e2e test only when a transport adapter is built. Think I can just remove it.